### PR TITLE
Add cli option to broadcast to 255.255.255.255

### PR DESCRIPTION
**Pull request by rkost**
This adds a cli option to make the game controller broadcasting to 255.255.255.255. This can be useful for some network configurations.

### CODE REVIEW

File: src/controller/GameController.java [hunk 1]
```diff
@@ -318,7 +330,7 @@ public static void main(String[] args) {
         Log.toFile("Competition phase = " + (data.competitionPhase == GameControlData.COMPETITION_PHASE_PLAYOFF ? "playoff" : "round robin"));
         Log.toFile("Competition type = "  + (data.competitionType == GameControlData.COMPETITION_TYPE_GENERAL_PENALTY_KICK ? "general penalty kick challenge" : "normal"));
         Log.toFile("Auto color change = " + data.colorChangeAuto);
-        Log.toFile("Using broadcast address " + (localAddress.getBroadcast() == null ? localAddress.getAddress() : localAddress.getBroadcast()));
+        Log.toFile("Using broadcast address " + broadcastAddress.toString());
```
Comment (jayenashar) [inline src/controller/GameController.java:333]: does it not automatically call `.toString()`?
```suggestion
        Log.toFile("Using broadcast address " + broadcastAddress);
```

File: src/controller/GameController.java [hunk 2]
```diff
@@ -275,13 +283,17 @@ public static void main(String[] args) {
             Rules.league.delayedSwitchAfterGoal = 0;
         }
 
+        InetAddress broadcastAddress = localAddress.getBroadcast() == null ? localAddress.getAddress() : localAddress.getBroadcast();
```
Comment (jayenashar) [inline src/controller/GameController.java:286]: can this be made final?
```suggestion
        final InetAddress broadcastAddress = limitedBroadcast ? InetAddress.getByName("255.255.255.255") : localAddress.getBroadcast() == null ? localAddress.getAddress() : localAddress.getBroadcast();
```

**Review by jayenashar — APPROVED** [review]
i'm not sure i understand why this is a "limited" broadcast

File: src/controller/GameController.java (hunk 2 above)
Comment (ahasselbring) [reply src/controller/GameController.java:286]: `InetAdress.getByName` can throw an `UnknownHostException`, so not without moving this and also the Log prints into a common try-catch-block.

**Comment by rkost** [comment]
> i'm not sure i understand why this is a "limited" broadcast

Naming is a bit strange for broadcast addresses imo. See https://de.wikipedia.org/wiki/Broadcast and https://www.juniper.net/documentation/en_US/junose15.1/topics/concept/ip-broadcast-addressing-overview.html

File: src/controller/GameController.java (hunk 1 above)
Comment (rkost) [reply src/controller/GameController.java]: Ah, you are right. Force pushed

**Pull request merged**